[FEATURE] MCP Client: allow customizing JSON ser/deser

The current implementation of `langchain4j-mcp` fails on deserialization, when a MCP Server is already implementing the latest 2025-11-25 specification. For instance, the static ObjectMapper instances fail to deserialize due to failing on unknown properties (i.e. Resource Template `icons` attribute).

**Describe the bug**
The static ObjectMapper of the DefaultMcpClient Implementation seems to be configured in the default settings, not allowing unknown fields to be present in the response json.

In the recent MCP specification `2025-11-25`, new (imo backwards compatible) fields were introduced to some responses, e.g. Resource Templates and Resources are containing `icons`.

The current object model of those is not containing those yet and thus the deserialization failure.

**Log and Stack trace**
```
Caused by: java.lang.IllegalArgumentException: Unrecognized field "icons" (class dev.langchain4j.mcp.client.McpResourceTemplate), not marked as ignorable (4 known properties: "name", "description", "uriTemplate", "mimeType"])
 at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: dev.langchain4j.mcp.client.McpResourceTemplate["icons"])
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4730)
	at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:4661)
	at dev.langchain4j.mcp.client.ResourcesHelper.parseResourceTemplateRefs(ResourcesHelper.java:73)
	at dev.langchain4j.mcp.client.DefaultMcpClient.obtainResourceTemplateList(DefaultMcpClient.java:400)
	at dev.langchain4j.mcp.client.DefaultMcpClient.listResourceTemplates(DefaultMcpClient.java:348)
        ...
```

**To Reproduce**

* Use `langchain4j-mcp` in version `1.9.1`
* Connect to Remote Github MCP server with Personal Access Token
  * see https://github.com/github/github-mcp-server/blob/main/docs/remote-server.md
* Fetch Resource Templates using the MCP client

**Expected behavior**
In my opinion, the latest MCP spec contains backwards compatible changes to the Response Model and thus, the McpClient implementation should not break on such changes, rather than ignoring those fields and ideally yielding a warning.

Furthermore, the ObjectMapper instances should be configurable, or one should get the capability to inject a custom defined ObjectMapper instance to configure Serdes features on their own.

**Please complete the following information:**
- LangChain4j version: 1.9.1(-beta17)
- LLM(s) used: not relevant
- Java version: 25
- Spring Boot version (if applicable): 4.0.0

--- a/langchain4j-mcp/src/main/java/dev/langchain4j/mcp/client/DefaultMcpClient.java
+++ b/langchain4j-mcp/src/main/java/dev/langchain4j/mcp/client/DefaultMcpClient.java
@@ -5,6 +5,7 @@ import static dev.langchain4j.internal.Utils.isNullOrBlank;
 import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.node.ArrayNode;
@@ -53,9 +54,12 @@ import org.slf4j.LoggerFactory;
 public class DefaultMcpClient implements McpClient {
 
     private static final Logger log = LoggerFactory.getLogger(DefaultMcpClient.class);
+
+    static final ObjectMapper OBJECT_MAPPER = new ObjectMapper()
+            .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+
     private final AtomicLong idGenerator = new AtomicLong(0);
     private final McpTransport transport;
-    static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
     private final String key;
     private final String clientName;
     private final String clientVersion;

